src/blocks/uptime.rs

```diff
@@ -0,0 +1,137 @@
+            tx_update_request: tx_update_request,
+            config: config,
+        })
+    }
+}
+
+fn read_file(path: &str) -> Result<String> {
+    let mut f = OpenOptions::new()
+        .read(true)
+        .open(path)
+        .block_error("uptime", &format!("failed to open file {}", path))?;
+    let mut content = String::new();
+    f.read_to_string(&mut content)
+        .block_error("uptime", &format!("failed to read {}", path))?;
+    // Removes trailing newline
+    content.pop();
+    Ok(content)
+}
+
+impl Block for Uptime {
+    fn update(&mut self) -> Result<Option<Duration>> {
+        let uptime_raw = match read_file("/proc/uptime") {
+            Ok(file) => file,
+            Err(e) => {
+                return Err(BlockError(
+                    "Uptime".to_owned(),
+                    format!("Uptime failed to read /proc/uptime: '{}'", e),
+                ));
+            }
+        };
+        let uptime = match uptime_raw.split_whitespace().nth(0) {
+            Some(uptime) => uptime,
+            None => {
+                return Err(BlockError(
+                    "Uptime".to_owned(),
+                    "Uptime failed to read uptime string.".to_owned(),
+                ));
+            }
+        };
+
+        let total_seconds = match uptime.parse::<f64>() {
+            Ok(uptime) => uptime as u32,
+            Err(e) => {
+                return Err(BlockError(
+                    "Uptime".to_owned(),
+                    format!("Uptime failed to convert uptime float to integer: '{}')", e),
+                ));
+            }
+        };
+
+        // split up seconds into more human readable portions
+        let weeks = (total_seconds / 604_800) as u32;
+        let rem_weeks = (total_seconds % 604_800) as u32;
+        let days = (rem_weeks / 86_400) as u32;
+        let rem_days = (rem_weeks % 86_400) as u32;
+        let hours = (rem_days / 3600) as u32;
+        let rem_hours = (rem_days % 3600) as u32;
+        let minutes = (rem_hours / 60) as u32;
+        let rem_minutes = (rem_hours % 60) as u32;
+        let seconds = rem_minutes as u32;
+
+        let mut text = String::from("0");
+        // display two digits at most
+        if hours == 0 && days == 0 && weeks == 0 {
+            text = format!("⏱ {}m {}s", minutes, seconds);
+        } else if hours > 0 && days == 0 {
+            text = format!("⏱ {}h {}m", hours, minutes,);
+        } else if hours > 0 && days > 0 && weeks == 0 {
+            text = format!("⏱ {}d {}h", days, hours);
+        } else if days > 0 && weeks > 0 {
+            text = format!("⏱ {}w {}d", weeks, days);
+        }
+        //debug:  let text = format!("uptime: {}w, {}d, {}h, {}m, {}s", weeks, days, hours, minutes, seconds);
+        self.text.set_text(text);
+        Ok(Some(self.update_interval))
+    }
+
+    fn view(&self) -> Vec<&I3BarWidget> {
+        vec![&self.text]
+    }
+
+    fn click(&mut self, _: &I3BarEvent) -> Result<()> {
```

Comment:
Remove the handler entirely. It has a default implementation,

Reply:
Ok, I copied most of this from template.rs